src/main/java/uk/co/ramp/covid/simulation/place/Household.java

```diff
@@ -178,4 +186,106 @@ public void cycleMovements(int day, int hour, boolean lockdown) {
         }
         people.removeAll(left);
     }
+
+    @Override
+    public void doMovement(int day, int hour, boolean lockdown) {
+       moveShift(day, hour, lockdown);
+
+       // Implies shopping trips have higher priority than neighbour and restaurant trips
+       moveShop(day, hour, lockdown);
+
+       moveNeighbour(day, hour);
+       if (!lockdown) {
+           moveRestaurant(day, hour);
+       }
+       
+       sendNeighboursHome(day, hour);
+    }
+
+    private void moveNeighbour(int day, int hour) {
+        List<Person> left = new ArrayList();
+
+        for (Household n : getNeighbours()) {
+            if (RNG.get().nextUniform(0, 1) < PopulationParameters.get().getNeighbourVisitFreq()) {
+                // We visit neighbours as a family
+                for (Person p : getInhabitants()) {
+                    if (!p.getQuarantine()) {
+                        n.addPersonNext(p);
+                        left.add(p);
+                    }
+                }
+                break;
+            }
+        }
+        
+        people.removeAll(left);
+    }
+
+    private void moveShop(int day, int hour, boolean lockdown) {
+        List<Person> left = new ArrayList();
+
+        double visitProb = PopulationParameters.get().getpGoShopping();
+        if (lockdown) {
+            visitProb = visitProb * 0.5;
+        }
+
+        if (RNG.get().nextUniform(0, 1) < visitProb) {
+            Shop s = places.getRandomShop();
+            if (s == null) {
+                return;
+            }
+            // We go shopping as a family
+            if (s.isVisitorOpenNextHour(day, hour)) {
```

Comment:
Do all family members always go shopping together? Do we need to some way to decide how many members of a family go on each shopping trip?

Reply:
Paul suggests (for now at least) they all go and return to/from restaurants/shops/neighbours together.